### PR TITLE
Update and improve Component documentation

### PR DESCRIPTION
**Pull request by locks**

### CODE REVIEW

**Comment by tomdale** [comment]
@locks Looks like Travis is failing due to trailing whitespace.

**Comment by rwjblue** [comment]
@locks  - Ping!

**Comment by locks** [comment]
@tomdale g2g

File: packages/@glimmer/component/src/tracked.ts [hunk 1]
```diff
@@ -12,6 +12,55 @@ import { dict, Dict } from "@glimmer/util";
...
+ * ```typescript
+ * import Component, { tracked } from '@glimmer/component';
+ *
+ * export default class MyComponent extends Component {
+ *    @tracked
+ *    remainingApples = 10
+ * }
+ * ```
+ *
+ * When something changes the component's `remaining` property, the rerender
```
Comment (Turbo87) [inline packages/@glimmer/component/src/tracked.ts]: this should be `remainingApples` instead of `remaining`, right?

File: packages/@glimmer/component/src/tracked.ts [hunk 2]
```diff
@@ -12,6 +12,55 @@ import { dict, Dict } from "@glimmer/util";
...
+ * ```typescript
+ * import Component, { tracked } from '@glimmer/component';
+ *
+ * export default class MyComponent extends Component {
+ *   totalApples = 100
+ *
+ *   @tracked
+ *   eatenApples = 0
+ *
+ *   @tracked('totalApples', 'eatenApples')
```
Comment (Turbo87) [inline packages/@glimmer/component/src/tracked.ts]: this seems strange. why is `totalApples` mentioned as a dependency, but `totalApples` above is not marked as `@tracked`?

File: packages/@glimmer/component/src/tracked.ts [hunk 3]
```diff
@@ -194,7 +243,7 @@ function hasOwnProperty(obj: any, key: symbol) {
   return hOP.call(obj, key);
 }
 
-let propertyDidChange = function() {};
+let propertyDidChange = function () { };
```
Comment (Turbo87) [inline packages/@glimmer/component/src/tracked.ts]: unrelated whitespace changes?

Comment (locks) [reply packages/@glimmer/component/src/tracked.ts]: ugh, prettier

File: packages/@glimmer/component/src/tracked.ts (hunk 2 above)
Comment (locks) [reply packages/@glimmer/component/src/tracked.ts]: I think `totalApples` can be removed from the dependencies.

Comment (Turbo87) [reply packages/@glimmer/component/src/tracked.ts]: we then need to explicitly explain it though that it does not have to be mentioned because it itself is not `@tracked` and probably even why it is not `@tracked`

File: packages/@glimmer/component/src/tracked.ts [hunk 4]
```diff
@@ -12,6 +12,55 @@ import { dict, Dict } from "@glimmer/util";
...
+ * ```typescript
+ * import Component, { tracked } from '@glimmer/component';
+ *
+ * const totalApples = 100;
+ *
+ * export default class MyComponent extends Component {
+ *    @tracked
+ *    eatenApples = 0
+ *
+ *    @tracked('eatenApples')
```
Comment (t-sauer) [inline packages/@glimmer/component/src/tracked.ts]: Might be helpful to provide an example that has multiple dependents.

File: packages/@glimmer/component/src/tracked.ts [hunk 5]
```diff
@@ -12,6 +12,55 @@ import { dict, Dict } from "@glimmer/util";
...
+ * When something changes the component's `remainingApples` property, the rerender
+ * will be scheduled.
+ *
+ * @example Dependents
+ *
+ * In the case that you have a computed property that depends other
+ * properties, you want to track both so that when one of the
+ * dependents change, a rerender is scheduled.
+ *
+ * In the following example we have three properties, `totalApples`,
```
Comment (jmbejar) [inline packages/@glimmer/component/src/tracked.ts]: `totalApples` is not a property of the component class. I would change it to:

> In the following example we have two properties,`eatenApples` and `remainingApples`.
